core/src/main/java/org/elasticsearch/index/query/BoostingQueryBuilder.java

```diff
@@ -172,15 +172,15 @@ public Query toQuery(QueryParseContext parseContext) throws QueryParsingExceptio
 
     @Override
```

Comment:
this is fishy. I think part of what we are doing here is make sure we throw the same errors in the same situations. Before our refactoring everything would go through the parsers, thus validation in the parser rules. After the refactoring the parser can be skipped, meaning that we can't have validation that happens only in the parsers and is skipped in the builders, that is why we have the validate method. Having doXContent producing something that the parser won't accept is a no-go. I think we should go over all the refactored parsers and double check validation that happens there and can be moved to the builder for consistency. Makes sense?